### ISSUE TITLE
AbstractControl pristine and dirty properties behaviour

### ISSUE THREAD

**Issue by fveltmann** [angular/angular#46410]
### Description

The properties _pristine_ and _dirty_ should indicate if the value of a FormControl or FormGroup has changed yet, but the behaviour of those properties is not perfectly logical. If you start typing another value into a FormControl the value has changed referenced to the initial value. But if you type the inital value back into the control, the properties remain and that is not logical.
In that case you have only touched the control and the properties _touched_ and _untouched_ are set correctly.

### Proposed solution

_Pristine_ and _dirty_ should be chnaged only when the value of a control is different to it's initial value. Currently pristine and dirty indicate pretty much the same when you have change something in the control, but revert the changes.

### Alternatives considered

Alternatively some additional properties that indicate the behaviour I described would be useful. E.g. for a case when you want to show a save-bar but only when values are different to the initial values.

**Comment by fveltmann** [comment]
I agree that changing the current logic is not the right way, because it would be a breaking change.
But I think adding new attributes for this case would be very useful for some scenarios and not a big deal.

**Comment by jcmalek** [comment]
I agree completely that Angular's definition of "dirty" is not what I would have expected or what is necessarily useful to me. There are also others who are clearly experiencing the same confusion:
https://stackoverflow.com/questions/41022263/angular2-formcontrol-stays-dirty-even-if-set-to-original-value

I also understand that it would probably be a huge breaking change to modify this one at this point. However, adding a new property for this could be handy for a ton of people so we all don't have to re-implement this independently and repeatedly.

Since "dirty" is unfortunately taken, I would perhaps recommend a name like "default" or "isDefault" for a boolean property indicating if the form values match the default values.

(It could also be nice if we had a mechanism to change a control's default values after creation without having to reconstruct the entire form control tree from scratch but that is perhaps another issue/ticket)
